### PR TITLE
Issue 1221 - tiny fix

### PR DESCRIPTION
**Pull request by o-smirnov**

### CODE REVIEW

**Comment by o-smirnov** [comment]
Hmmm, actually not so tiny, I forgot about the previous commits.

**Comment by o-smirnov** [comment]
OK, so this also adds the following options:

* ``selfcal: img_mfs_weighting``, to control the MFS weighting option (default on)

* ``crosscal: spw_g`` and ``spw_k``, to select a subset of the band for FG and K calibration (previous option -- not used by anybody, to my knowledge -- was ``spw`` and applied to all three, but since K can benefit from a wider chunk of band, IK decoupled them)

There is a potential breakage here if we have anyone using the ``crosscal: spw`` setting. We could consider supporting in  a backwards-compatible way (i.e. use ``spw`` for all three terms if supplied, override with ``spw_g`` and ``spw_k`` if supplied), at the cost of complicating the config.

**Comment by KshitijT** [comment]
With this from the wsclean wiki:
```
MF weighting improves the frequency-integrated (multi-frequency, "MF") images. For science using the spectral images, MF weighting is discouraged. This is because MF weighting adjusts the weight of individual frequencies to make the integrated image look best. This however can create artificial spectral structure when the individual frequency images are inspected / used in science. It is still possible to use -join-channels for those science cases, but make sure MF weighting is turned off with -no-mf-weighting.
```
I am a bit scared about this particular parameter and would argue that `img_mfs_weighting` default should be `false`, cute images be damned. What do you think, @o-smirnov ?

File: caracal/schema/selfcal_schema.yml [hunk 1]
```diff
@@ -143,7 +148,7 @@ mapping:
         example: 'True'
       img_specfit_nrcoeff:
         type: int
-        desc: Number of spectral polynomial terms to fit to each clean component. This is equal to the order of the polynomial plus 1.
+        desc: Number of spectral polynomial terms to fit to each clean component. This is equal to the order of the polynomial plus 1. Use -1 to disable fitting.
```
Comment (KshitijT) [inline caracal/schema/selfcal_schema.yml:151]: @o-smirnov , could you please add a line here saying that with  `img_specfit_nrcoeff` set to -1, Crystalball won't be able to work?

**Comment by o-smirnov** [comment]
Yeah, now that you put it this way....

**Review by KshitijT — CHANGES_REQUESTED** [review]
Just the one line change, otherwise LGTM.

**Comment by paoloserra** [comment]
I thought in https://github.com/caracal-pipeline/caracal/issues/1221#issuecomment-669163719 we agreed that 0 indicates no fitting (you're fitting 0 terms), and there is no need of a -1 option. Have I missed something?

File: caracal/schema/selfcal_schema.yml (hunk 1 above)
Comment (paoloserra) [reply caracal/schema/selfcal_schema.yml:151]: Please see my comment on the -1 option not actually being necessary. In. the code -1 does the same as 0 (0 terms fitted). So I think the desc in the schema should replace -1 with 0.

**Comment by o-smirnov** [comment]
ok to test

**Comment by o-smirnov** [comment]
@KshitijT @paoloserra please check these last minor changes and give your seal of approval.

**Review by KshitijT — APPROVED** [review]

**Comment by KshitijT** [comment]
LGTM. As an aside, I am a bit surprised that we were not using deconvolution-channels option.

**Review by paoloserra — APPROVED** [review]

**Comment by paoloserra** [comment]
> LGTM. As an aside, I am a bit surprised that we were not using deconvolution-channels option.

Which option is that? Are we using it now?

**Comment by KshitijT** [comment]
> > LGTM. As an aside, I am a bit surprised that we were not using deconvolution-channels option.
> 
> Which option is that? Are we using it now?

No, apparently we haven't.
The option is here in stimela: https://github.com/ratt-ru/Stimela/blob/master/stimela/cargo/cab/wsclean/parameters.json#L520 . Used for saving memory.

The reason I was looking at it is that if now img_specfit_nrcoeff is switched off, then it wsclean would crash if deconvolution-channels was activated. But we don't have the parameter, so no need to worry, I guess.

**Comment by KshitijT** [comment]
Ready to be merged. @o-smirnov , you want to do the honours? ;)

**Pull request merged**